vector/src/main/java/im/vector/app/features/roomdirectory/PublicRoomsViewState.kt

```diff
@@ -21,19 +21,22 @@ import com.airbnb.mvrx.MvRxState
 import com.airbnb.mvrx.Uninitialized
 import org.matrix.android.sdk.api.session.room.members.ChangeMembershipState
 import org.matrix.android.sdk.api.session.room.model.roomdirectory.PublicRoom
+import org.matrix.android.sdk.api.session.room.model.thirdparty.ThirdPartyProtocol
 
 data class PublicRoomsViewState(
         // The current filter
         val currentFilter: String = "",
         // Store cumul of pagination result
-        val publicRooms: List<PublicRoom> = emptyList(),
+        val publicRooms: Map<PublicRoom, RoomDirectoryData> = emptyMap(),
         // Current pagination request
         val asyncPublicRoomsRequest: Async<Unit> = Uninitialized,
-        // True if more result are available server side
-        val hasMore: Boolean = false,
         // Set of joined roomId,
         val joinedRoomsIds: Set<String> = emptySet(),
         // keys are room alias or roomId
         val changeMembershipStates: Map<String, ChangeMembershipState> = emptyMap(),
-        val roomDirectoryData: RoomDirectoryData = RoomDirectoryData()
+        val roomDirectoryData: RoomDirectoryData = RoomDirectoryData(),
```

Comment:
Don't you think we should remove this field to clean up the state?

Reply:
Yes. I hesitated to do that. Applied